Add publishing info to project.

sjoerdtalsma opened the pull request:
- Add maven wrapper
- Add various maven plugins
- Add CONTRIBUTING.md and RELEASE.md
- Add travis configuration
- Sort out credentials

sjoerdtalsma committed:
Add maven wrapper to project.

sjoerdtalsma committed:
Add license maven plugin.

sjoerdtalsma committed:
Add release + centralsync maven plugins.

sjoerdtalsma committed:
Add issueManagement + distributionManagement to pom.

sjoerdtalsma committed:
Add CONTRIBUTING.md.

sjoerdtalsma committed:
Fix typo in CONTRIBUTING.md.

sjoerdtalsma committed:
Add maven settings and travis files.

sjoerdtalsma commented:
Requested sontatype access:
https://issues.sonatype.org/browse/OSSRH-19982?focusedCommentId=389593&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-389593

sjoerdtalsma commented:
@yurishkuro @bensigelman 
Could you add me to bintray?
My profile is https://bintray.com/sjoerdtalsma

https://bintray.com/opentracing/organization/edit

codefromthecrypt commented on pom.xml:
```diff
@@ -124,6 +196,60 @@
+                <configuration>
+                    <subject>opentracing</subject>
+                    <repo>maven</repo>
+                    <packageName>opentracing-java</packageName>
```
this should probably be "java-globaltracer" (which would need to be created in bintray). otherwise you can end up with overlapping and/or confused released artifacts.

https://bintray.com/opentracing/maven

cc @ploffay as I also noticed there is none created for the jaxrs thing you made either

sjoerdtalsma replied:
cc @pavolloffay I think you were mentioned :wink:

pavolloffay replied:
:100:  thanks I forgot this too.

pavolloffay replied:
Has to be `packageName` equal to `artifactId`?

sjoerdtalsma replied:
Changed `packageName` to `java-globaltracer`.

codefromthecrypt reviewed (commented):
I'm assuming travis is switched on for this repo here: https://travis-ci.org/profile/opentracing-contrib if not, ask ben or yuri to click!

* make sure credentials are re-encrypted in travis yaml *

then, once bintray is sorted, should work!

yurishkuro commented:
> @yurishkuro @bensigelman
> Could you add me to bintray?
> My profile is https://bintray.com/sjoerdtalsma
> https://bintray.com/opentracing/organization/edit

I don't think I can, not seeing any admin tools on my account.

pavolloffay commented:
You can request an access directly there. (there is a button for it :)).

On Thu, Jan 12, 2017 at 5:16 PM, Yuri Shkuro <notifications@github.com>
wrote:

> @yurishkuro <https://github.com/yurishkuro> @bensigelman
> <https://github.com/bensigelman>
> Could you add me to bintray?
> My profile is https://bintray.com/sjoerdtalsma
> https://bintray.com/opentracing/organization/edit
>
> I don't think I can, not seeing any admin tools on my account.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/opentracing-contrib/java-globaltracer/pull/6#issuecomment-272206709>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFW6-KtsOEUjzZCStNgKA4GxnKYnk2fXks5rRlHPgaJpZM4Lhs3x>
> .
>

bhs commented on .travis.yml:
```diff
@@ -0,0 +1,53 @@
+env:
+  global:
+  # Ex. travis encrypt BINTRAY_USER=your_github_account
+  - secure: "QHoScwXVjj9mTZJuQ59jFvPr7Ri6tVgUyydk8Ba8CYDdgdNfnRzs3PW55g5iuC8Q8DJG6dsNfwWtYCagD8XIZw3bVE4Ic9hJJeOrY7PbaoB9gFb/emRYk83Z91xoYMXtaRCiOsxnXqMeyPd4rAc0mNOL5N00DjGVsDJp7BX4tAz0AL6mF4FNIkuuBIh7EEDpOJPSm4NXjUICaDQ8LETIVN69XvbVyoCu0lA2XIeSkXMq8O3izon4bch+fXe15z3LoWQQlmeK0YlEUZifPlIabIWErHEmjVrMVanIJu8B5SIF58pMhZ3AGL8PRBszDjMS+847LWCvW/VHsL9Le8pYXIxXksKw+DmioxudXV5jugDVy2qKsSIlUkP4MVFPwC4Pkt ...
```
@sjoerdtalsma since these strings are not exactly self-describing, I want to confirm that you have generated them with your creds using the travis cmdline tools.

sjoerdtalsma replied:
Not yet, but don't worry; I will before merging!

bhs replied:
alright -- let me know when that's happened. I don't have any concerns that haven't been brought up by others.

sjoerdtalsma replied:
Done

bhs commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,31 @@
+# Contributing to OpenTracing
```
nit: make this about the globaltracer specifically (that, or just do a redirect to the main OT-contrib guidelines.

sjoerdtalsma replied:
Done

bhs reviewed (commented):
LGTM assuming the credentials are correct (and that the other boilerplate/scripts here are copy-pasted from a reasonable source!).

sjoerdtalsma committed:
Add personal travis credentials.

sjoerdtalsma committed:
Fix groupId and centralSync packageName.

sjoerdtalsma committed:
Fix repo name in CONTRIBUTING.md.

sjoerdtalsma committed:
Add execute flag to ./mvnw

sjoerdtalsma committed:
Add execute flag to travis/publish.sh

sjoerdtalsma commented:
Shall I merge this?

bhs commented:
@sjoerdtalsma if you have a way of testing that the travis stuff will work end-to-end without merging, please do so. If not, let's wait 24h for people in other TZs then merge if there are no additional concerns. Thanks!

codefromthecrypt commented:
>
> I don't think I can, not seeing any admin tools on my account.
>
try here? https://bintray.com/opentracing/organization/edit

bhs commented:
(bintray: "bad UI" or "worst UI"???)

bhs commented:
@sjoerdtalsma I think you have to click "edit" by the OT profile then make your way here:

![image](https://cloud.githubusercontent.com/assets/2251553/21919998/3e333fb6-d913-11e6-847c-d74b44084c77.png)


Something like that. Def not intuitive.

sjoerdtalsma commented:
I think I'm member of the opentracing bintray organization.
Do I have to pre-create the 'maven' package for `java-globaltracer` beforehand or will travis take care when it publishes (since my token is encoded in the .travis.yml already)?
@adriancole ?

Sorry guys, this is my first time using both travis and bintray.

wu-sheng commented:
👏 @sjoerdtalsma , I think you can help me on the other repo in opentracing-contrib, later. 
Waiting for your success, haha. 😝

sjoerdtalsma committed:
Re-encrypt travis secrets to correct repo pubkey.

They were encrypted to the public-key of my forked repository.
The main repository has a different public-key.
This can be found out calling `travis pubkey`
vs. `travis pubkey -r opentracing-contrib/java-globaltracer`.

sjoerdtalsma committed:
Fix bintray repository url.

sjoerdtalsma committed:
Add RELEASE document.

Referring to `java-opentracing` release documentation.

sjoerdtalsma committed:
Add CI + Maven shields.

bhs commented:
@sjoerdtalsma 

> Do I have to pre-create the 'maven' package for java-globaltracer beforehand or will travis take care when it publishes (since my token is encoded in the .travis.yml already)?

IIRC it will create the maven package on-demand.
